### PR TITLE
CSS-5118 Active status verification

### PR DESCRIPTION
**Pull request by AmberCharitos**
The `ActiveStatus` was previously being set immediately following pebble re-plan. This was misleading as the application has not been verified to be running before setting this status. This PR utilises [checks](https://juju.is/docs/sdk/interact-with-pebble#heading--check-configuration) to set the status as either Maintenance (unverified or unsuccessful verification) or Active (successful verification). 

A few notes:
- verification is via access to the web server hence this only applies when the superset charm has been configured with `app` or `gunicorn app` functions. Additional verification will be put in place for other functions with future PRs.
- The `update_status` hook is by default run every 5 minutes. In order to speed up the process of verification the `juju model-config` should be updated for this to run every minute. Otherwise the application cannot be marked as active for a minimum of 5 minutes. 
- on check failure being marked as `ignore` as opposed to `restart` is due to the fact that the application will show `CheckStatus.DOWN` while starting up the web server as this can take some time. And we do not want the application to restart due to this.

### CODE REVIEW

**Review by kelkawi-a — APPROVED** [review]
LGTM

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -62,13 +62,19 @@ juju model-config logging-config="<root>=INFO;unit=DEBUG"
 juju status
 juju debug-log
 ```
+### Configure juju
+```
+# customise update status hook interval:
+juju model-config update-status-hook-interval=1m
```
Comment (mertalpt) [inline CONTRIBUTING.md:68]: Does this not effect every application in the model? Were there any alternatives in which we did not have to mess with the fundamental defaults?

**Review by mertalpt — APPROVED** [review]
LGTM with a comment.

File: CONTRIBUTING.md (hunk 1 above)
Comment (AmberCharitos) [reply CONTRIBUTING.md:68]: It does, though this model will be specific to superset and related charms, for which I would also use this `check`. An alternative would be we leave the default as is and it's not possible for the charm to be marked as `Active` for a minimum of 5 minutes. Documentation could be updated to say deploy superset and wait 5 minutes for it to become active. I have seen similar in other charms. 

I will leave like this for now but if this becomes an issue for other charms in the model I will consider requiring the 5 minute wait. Unless you have an alternative suggestion in mind?

Comment (mertalpt) [reply CONTRIBUTING.md:68]: Maybe I do not fully understand what is going on here. That interval is a change to make the charm hooks more responsive, not a hard requirement, right? I have the feeling that charms should just work without messing with the underlying parameters. But that is just my opinion so if you think this is fine then go with it, I don't think it is a big problem.

Comment (AmberCharitos) [reply CONTRIBUTING.md:72]: I've added this as a note to the `CONTRIBUTING.md` and mentioned it is optional.

**Pull request merged**